### PR TITLE
Feature/fasteffekt list

### LINKED ISSUE #7: List

### PR DESCRIPTION
**Pull request by IR0NSIGHT**
fixes #7

### CODE REVIEW

File: src/list.effekt
Comment (phischu) [inline src/list.effekt:1]: Please indent by 2

File: src/list.effekt [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        }
+    }
+    return bigger;
+}
+
+def makeList(length: Int): List[Int] = {
+    if (length == 0) {
+        Nil()
+    } else {
+        Cons(length, makeList(length-1))
```
Comment (phischu) [inline src/list.effekt:29]:
```suggestion
        Cons(length, makeList(length - 1))
```

File: src/list.effekt [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    var bigger = false;
+    while (run) {   //TODO use recursion here
+        (xTail,yTail) match {
+            case (_, Nil()) => run = false; bigger = false;
+            case (Nil(), _) => run =false; bigger = true;
+            case (Cons(_,xx),Cons(_,yy)) => xTail = xx; yTail = yy;
+        }
+    }
+    return bigger;
+}
```
Comment (phischu) [inline src/list.effekt:23]:
```suggestion
def isShorterThan(x: List[Int], y: List[Int]): Boolean = (x, y) match {
  case (Nil(), Nil()) => false
  case (Nil(), Cons(_, _)) => true
  case (Cons(_, _), Nil()) => false
  case (Cons(_, xTail), Cons(_, yTail)) => isShorterThan(xTail, yTail)
}
```

File: src/list.effekt [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+                tail(
+                    tail(xChild, y, z),
+                    tail(yChild, z, x),
+                    tail(zChild, x, y)
+                )
+            case _ => panic("oh no!");
+        }
+    } else {
+        z;
```
Comment (phischu) [inline src/list.effekt:50]:
```suggestion
        z
```

File: src/list.effekt [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+                val yChild : List[Int] = yy;
+                val zChild : List[Int] = zz;
+                val xChild : List[Int] = xx;
+
+                tail(
+                    tail(xChild, y, z),
+                    tail(yChild, z, x),
+                    tail(zChild, x, y)
+                )
+            case _ => panic("oh no!");
```
Comment (phischu) [inline src/list.effekt:47]:
```suggestion
            case (Cons(_, xChild),Cons(_, yChild),Cons(_, zChild)) =>
                tail(
                    tail(xChild, y, z),
                    tail(yChild, z, x),
                    tail(zChild, x, y)
                )
            case _ => panic("oh no!");
```

File: src/list.effekt [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+def List(_: Int) = {
+    def benchmark(): Int = {
+        val result = 
+        tail(
+            makeList(15),
+            makeList(10),
+            makeList(6)
+        );
+        length(result);
```
Comment (phischu) [inline src/list.effekt:62]:
```suggestion
        val result = tail(
          makeList(15),
          makeList(10),
          makeList(6)
        );
        length(result)
```

**Review by phischu — CHANGES_REQUESTED** [review]

**Comment by IR0NSIGHT** [comment]
i messed up the merge, and had to hard reset master.
merge commit is 966634f4 sth sth

**Pull request merged**